lib/fika/compiler/type_checker.ex

```diff
@@ -460,21 +481,37 @@ defmodule Fika.Compiler.TypeChecker do
     end)
   end
 
+  defp set_latest_call(env, mfa) do
+    env
+    |> Map.put(:latest_called_function, mfa)
+    |> Map.put_new(:callstack, FunctionDependencies.new_graph())
```

Comment:
Whenever we start the callstack, we don't have the :callstack set, so we need to create a new dependency graph.
In a future refactor, we can use Env structs instead of plain maps, and force envs to have new graph by default.

Also, the first call in the stack will have `:latest_called_function` unset, so we currently check this to avoid touching an unset graph